pkg/standalone/standalone.go

```diff
@@ -55,6 +56,31 @@ func (h *delegateHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	w.WriteHeader(http.StatusNotFound)
 }
 
+// TODO: replace this with clientcmd
+func GetAPIServerClient(authPath string, apiServerList util.StringList) (*client.Client, error) {
```

Comment:
I'm not a fan of Die - for one thing, it means if someone doesn't want to die, they duplicate the method, instead of just handling error.  We should definitely not try to initialize it if apiserverlist is empty - would that work instead?

----- Original Message -----
> > @@ -55,6 +56,31 @@ func (h *delegateHandler) ServeHTTP(w
> > http.ResponseWriter, req *http.Request) {
> >  	w.WriteHeader(http.StatusNotFound)
> >  }
> >  
> > +// TODO: replace this with clientcmd
> > +func GetAPIServerClient(authPath string, apiServerList util.StringList)
> > (*client.Client, error) {
> 
> suggest calling this GetAPIServerClientOrDie, and hoisting the check for
> apiServerList < 1 outside of this function so it is called
>   if len(apiServerList) > 0 {
>     cl = GetApiserverClientOrDie(...)
>   } else {
>     cl = nil
>   }
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/3270/files#r22610319